src/cloudWatchLogs/cloudWatchLogsUtils.ts

```diff
@@ -13,6 +13,15 @@ import { CloudWatchLogsParameters } from './registry/logStreamRegistry'
 // The following functions are used to structure and destructure relevant information to/from a URI.
 // Colons are not valid characters in either the group name or stream name and will be used as separators.
 
+/**
+ * This function concatenates the path and query to create a unique (enough) identifier for a URI.
+ * @param uri
+ * @returns
+ */
+export function uriToKey(uri: vscode.Uri): string {
+    return uri.path + uri.query
```

Comment:
queries can be identical semantically though the params are in a different order. if we want to treat the query as part of a key maybe we need to sort by param names here before putting it in string form.